package/MDAnalysis/analysis/base.py

```diff
@@ -40,18 +41,111 @@
 
 
+class Results(dict):
+    r"""Container object for storing results.
+
+    Results are dictionaries that provide two ways in which can values be
+    accessed: by dict key ``results["value_key"]`` or by object attribute,
+    ``results.value_key``. They store all results obatined from an analysis
+    after the ``run`` call.
+
+    The current implementation is similar to the :class:`sklearn.utils.Bunch`
+    class in `scikit-learn`_.
+
+    .. _`scikit-learn`: https://scikit-learn.org/
+
+
+    Examples
+    --------
+    >>> results = Results(a=1, b=2)
+    >>> results['b']
+    2
+    >>> results.b
+    2
+    >>> results.a = 3
+    >>> results['a']
+    3
+    >>> results.c = [1, 2, 3, 4]
+    >>> results['c']
+    [1, 2, 3, 4]
+
+
+    Raises
+    ------
+    ValueError
+        If a to assigned attribute has the same name as a default dictionary
+        attribute.
+
+    ValueError
+        If a key is not of type ``str`` and therefore is not able to be
+        accessed by attribute.
+    """
+
+    def _validate_key(self, key):
+        if key in dir(dict):
+            raise TypeError(f"'{key}' is a protected dictionary attribute")
+        elif not (isinstance(key, str) and key[0].isalpha()
+                  and re.match("^[a-zA-Z0-9_]*$", key)):
+            raise TypeError(f"'{key}' is not able to be accessed by attribute")
+
+    def __init__(self, **kwargs):
+        for key in kwargs.keys():
+            self._validate_key(key)
+        super().__init__(kwargs)
+
+    def __setattr__(self, key, value):
+        self._validate_key(key)
+        self[key] = value
+
+    def __setitem__(self, key, value):
+        self._validate_key(key)
+        super().__setitem__(key, value)
+
+    def __dir__(self):
+        return self.keys()
+
+    def __getattr__(self, key):
+        try:
+            return self[key]
+        except KeyError as err:
+            raise AttributeError("'Results' object has no "
+                                 f"attribute '{key}'") from err
+
+
 class AnalysisBase(object):
-    """Base class for defining multi frame analysis
+    r"""Base class for defining multi frame analysis
 
     The class it is designed as a template for creating multiframe analyses.
     This class will automatically take care of setting up the trajectory
     reader for iterating, and it offers to show a progress meter.
+    Computed results are stored inside the :attr:`results` attribute.
 
     To define a new Analysis, `AnalysisBase` needs to be subclassed
-    `_single_frame` must be defined. It is also possible to define
-    `_prepare` and `_conclude` for pre and post processing. See the example
-    below.
+    :meth:`_single_frame` must be defined. It is also possible to define
+    :meth:`_prepare` and :meth:`_conclude` for pre and post processing.
+    All results should be stored as attributes of the :class:`Results`
+    container.
+
+    Parameters
+    ----------
+    trajectory : mda.Reader
```

Comment:
Could you be clearer with the class path here -- like MDAnalysis.coordinates.base.ReaderBase ?